spokestack/extensions/webrtc/agc.pyx

```diff
@@ -0,0 +1,81 @@
+"""
+This module contains the AutomaticGainControl class implemented in Cython
+"""
+# distutils: sources = filter_audio/agc/analog_agc.c, filter_audio/agc/digital_agc.c
+# distutils: include_dirs = filter_audio/agc/include/
+cimport numpy as np
+cimport cagc
+
+from spokestack.extensions.webrtc import ProcessError
+
+MIC_MAX = 255
+MIC_TARGET = 180
+
+
+cdef class WebRtcAgc:
+    cdef void* _agc
+
+    def __dealloc__(self):
+        cagc.WebRtcAgc_Free(self._agc)
+
+    def __init__(self,
+                 sample_rate,
+                 frame_width,
+                 target_level_dbfs,
+                 compression_gain_db,
+                 limit_enable
+    ):
+        self._agc = NULL
+        result = cagc.WebRtcAgc_Create(&self._agc)
+        if result != 0:
+            raise ValueError("invalid_config")
+
+        result = cagc.WebRtcAgc_Init(self._agc,
+                                     minLevel=0,
+                                     maxLevel=MIC_MAX,
+                                     agcMode=2,
+                                     fs=sample_rate
+        )
+        if result != 0:
+            raise ValueError("invalid_config")
+
+        cdef cagc.WebRtcAgc_config_t config
+        config.targetLevelDbfs = target_level_dbfs
+        config.limiterEnable = limit_enable
+        config.compressionGaindB = compression_gain_db
+
+        result = cagc.WebRtcAgc_set_config(self._agc, config)
+        if result != 0:
+            raise MemoryError
```

Comment:
this should probably be an invalid config error, since you're passing it configuration parameters, and allocations were probably done in create